src/account/AccountFactory.sol

```diff
@@ -63,6 +72,29 @@ contract AccountFactory is Ownable {
         return UpgradeableModularAccount(payable(addr));
     }
 
+    function createSemiModularAccount(address owner, uint256 salt) external returns (SemiModularAccount) {
+        // both module address and entityId for fallback validations are hardcoded at the maximum value.
+        bytes32 fullSalt = getSalt(owner, salt, type(uint32).max);
+
+        bytes memory immutables = _getImmutableArgs(owner);
+
+        address addr = _getAddressFallbackSigner(immutables, fullSalt);
+
+        // LibClone short-circuits if it's already deployed.
+        (bool alreadyDeployed, address instance) =
+            LibClone.createDeterministicERC1967(address(SEMI_MODULAR_ACCOUNT_IMPL), immutables, fullSalt);
+
+        if (instance != addr) {
+            revert SemiModularAccountAddressMismatch(addr, instance);
+        }
```

Comment:
Is it possible to trigger this case in a test? If not, could we remove this as an unnecessary check and just make the assertions between `getAddressFallbackSigner` and the actually deployed accounts in a test file?

Reply:
My thoughts exactly, I kept this for consistency but talked about removing it a bit, will dig into it a little before clearing it out!

Reply:
Removed in [4ab5b75](https://github.com/erc6900/reference-implementation/pull/123/commits/4ab5b75c3a64860510a53221ded7dd94564d8ee1)! As expected, I don't think it's possible to cause a revert here. We may want to consider adding invariants for this at some point-- but that'd be testing LibClone more than anything (which is fine!).